src/wm_info_provider/hyprland.rs

```diff
@@ -114,19 +114,25 @@ fn hyprland_cb(conn: &mut Connection<State>, state: &mut State) -> io::Result<()
         match hyprland.ipc.next_event() {
             Ok(event) => {
                 if let Some(active_ws) = event.strip_prefix("workspace>>") {
-                    hyprland.active_id = active_ws
-                        .parse()
-                        .map_err(|e| io::Error::new(io::ErrorKind::InvalidData, e))?;
-                    updated = true;
+                    match hyprland.workspaces.iter().find(|ws| ws.name == active_ws) {
+                        Some(ws) => {
+                            hyprland.active_id = ws.id;
+                            updated = true;
+                        }
+                        None => return Err(io::Error::new(io::ErrorKind::InvalidData, "Unknown workspace"))
+                    }
```

Comment:
Just a stylistic suggestion: I would prefer to use `.find(...).ok_or_else(|| ...)?` in this case.